fnet/network.go

```diff
@@ -34,6 +35,91 @@ func NormalizePort(port string) string {
 	return ":" + port
 }
 
+// BracketizeIPv6Address returns s in brackets if s is an IPv6 address.
+func BracketizeIPv6Address(s string) string {
```

Comment:
I don't think this function is needed, a url with ipv6 without [] is invalid (can't tell if last part is a port or part of the ip)

Reply:
That's originally what I expected too. I was surprised to find that `url.Parse` did not return an error when an IPv6 address was unbracketed within a url. See: https://play.golang.org/p/jqNv_1pnqvb

However, this is a moot point since I have refactored the code and dropped both func's that I originally added for this PR. Now that I have a better understanding of the requirements, I am simply updating the `grpcDestination` func with the necessary logic.